crates/testing/src/demo.rs

```diff
@@ -0,0 +1,510 @@
+//! Implementations for examples and tests only
+use commit::{Commitment, Committable};
+use derivative::Derivative;
+
+use hotshot::traits::election::static_committee::{GeneralStaticCommittee, StaticElectionConfig};
+
+use hotshot_signature_key::bn254::BLSPubKey;
+use hotshot_types::{
+    data::{fake_commitment, BlockError, ViewNumber},
+    traits::{
+        election::Membership,
+        node_implementation::NodeType,
+        state::{ConsensusTime, TestableState},
+        BlockPayload, State,
+    },
+};
+
+use serde::{Deserialize, Serialize};
+use std::{fmt::Debug, marker::PhantomData};
+
+use crate::demo::block::VIDTransaction;
+
+use self::block::{VIDBlockHeader, VIDBlockPayload};
+
+/// This module provides implementations of block traits for examples and tests only.
+pub mod block {
+    use std::{
+        fmt::{Debug, Display},
+        mem::size_of,
+    };
+
+    use ark_serialize::CanonicalDeserialize;
+    use commit::{Commitment, Committable};
+    use hotshot_types::{
+        data::{BlockError, VidCommitment, VidScheme, VidSchemeTrait},
+        traits::{
+            block_contents::{vid_commitment, BlockHeader, Transaction},
+            state::TestableBlock,
+            BlockPayload,
+        },
+    };
+    use serde::{Deserialize, Serialize};
+    use sha3::{Digest, Keccak256};
+
+    /// The transaction in a [`VIDBlockPayload`].
+    #[derive(Default, PartialEq, Eq, Hash, Serialize, Deserialize, Clone, Debug)]
+    pub struct VIDTransaction(pub Vec<u8>);
```

Comment:
Nit: Probably rename `VID*` structs to `Demo*`. "VID" used to indicate that they were associated with VID commitment, but since they've moved to `demo`, I think it's better to rename them.

Reply:
Done